### DISCUSSION TITLE
Postgres data not available on hobby plan after limits reached

### DISCUSSION THREAD

**Discussion by Ennoriel** [vercel/community#4437] [Help] [answered]
### Page to Investigate

https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing

### Steps to Reproduce

I have reached the limits on the hobby plan with Vercl Postgres. I cannot access my data anymore. The problem is: the data are my own. 

How can I download or make a dump of the database?

**Comment by amyegan** [comment, marked as answer]
You can use `pg_dump` and `pg_restore` to export and restore data: [vercel.com/docs/storage/vercel-postgres/faq#external-data-migration](https://vercel.com/docs/storage/vercel-postgres/faq#external-data-migration)

Vercel Postgres on a Hobby account is intended as a development database which enough functionality to let you create and test. Hobby accounts do not have the ability to exceed the [usage limits](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#pricing), but Pro accounts have the option to pay for additional usage on-demand. You can find more information about this on the Postgres Pricing page: [vercel.com/docs/storage/vercel-postgres/usage-and-pricing#hobby](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#hobby)

> Vercel will send you emails as you are nearing your usage limits. You will not pay for any additional usage. However, you will not be able to access Vercel Postgres if limits are exceeded. In this scenario, you will have to wait until 30 days have passed before using it again.

**Reply by alukach** [reply]
@amyegan To confirm, once a limit is reached (e.g. storage limit), the user is locked out of the hobby database for 30 days? 

> You can use `pg_dump` and `pg_restore` to export and restore data: [vercel.com/docs/storage/vercel-postgres/faq#external-data-migration](https://vercel.com/docs/storage/vercel-postgres/faq#external-data-migration)

This doesn't appear to be possible, as the limit lockout prevents this:

```
▶ pg_dump "postgres://default:password-removed@subdomain-removed.eu-central-1.aws.neon.tech:5432/verceldb?sslmode=require"

pg_dump: error: connection to server at "subdomain-removed.eu-central-1.aws.neon.tech" (1.2.3.4), port 5432 failed: ERROR:  Console request failed: endpoint is temporary unavailable. check your quotas and/or contact our support
```

~I'd prefer to be able to access the database to perform some delete operations to bring the storage back under limits. Is there no way to do this?~ Edit: upon further reading it appears that the storage limit is the _maximum_ storage used within a month ([source](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#storage)), so once that line is crossed you are put in timeout for the remainder of the month 😿 

At the risk of sounding dramatic, it does feel somewhat like a user's data is being held hostage without being given an opportunity to remedy the underlying limit issue. I respect that Vercel is a business and I have full intention of turning this account over to a paid account once this project goes live, however this lock-out experience leaves a bad taste in the mouth.
